### PR TITLE
fix(game): re-measure the HUD layout when its surface mounts

### LINKED ISSUE #5654: [Issue]: Game HUD never compacts when its surface mounts after the widget state hydrates
## Summary

When a Game chat's custom widget state hydrates while the transcript is still loading, `GameSurface` measures its HUD layout before the surface element exists and attaches no `ResizeObserver`, so the layout is never measured again on its own. At widths above the compact-by-width shortcut this leaves the desktop widget rails rendered on top of the dialogue until the window happens to be resized.

## Expected behavior

A Game chat with HUD widgets opened at a tablet or narrow-desktop width should collapse into the compact choice stage, with the widget rails flanking the choice cards, regardless of whether the transcript or the chat metadata resolved first.

## Actual behavior

The desktop widget rails render over the dialogue and the compact stage never appears. Nothing recovers it for the rest of the session except a window resize, because the observer that would have re-measured the layout was never attached.

The mechanism is in `packages/client/src/components/game/GameSurface.tsx`. The widget-layout effect is keyed on `[combatUiActive, normalizedWidgets.length]`. It measures `hudSurfaceRef.current` and attaches a `ResizeObserver` to it, but `GameSurface` also has an early return for its messages-loading branch that mounts no surface at all, and the widget state hydrates from chat metadata (`useSyncGameState`) independently of the messages query. When the widget count lands while that branch is showing, `updateWidgetLayout` finds no element and returns, and `const surface = hudSurfaceRef.current` is `null` so no observer is attached either. Neither dependency changes when the real surface mounts a moment later, so nothing re-measures.

#5618 changed that effect's reset from a flat `false` to `window.innerWidth < 768`, which makes the stranded value coincidentally correct on a phone. It does not address the measurement or the missing observer, so every case where the compact layout depends on the overlap estimate rather than raw width is still broken.

## Steps to reproduce

1. Open the app in a window about 900px wide.
2. Open a Game chat with an active session and four custom HUD widgets in its blueprint (two `hud_left`, two `hud_right`), whose latest assistant message carries a `[choices: ...]` block.
3. Arrange for the first page of messages to resolve after the chat detail, which is the ordinary ordering on a slow connection or a loaded machine. The reliable way to force it is to hold the first `GET /api/chats/<id>/messages` response until the surface has rendered its loading state.
4. Observe the widget rails sitting over the dialogue with no compact choice stage.
5. Resize the window by any amount and watch the layout correct itself.

Instrumenting the effect shows the two orderings directly:

```
messages resolve first (correct)
  effect widgets=4 surface=true   -> measured w=900 nextCompact=true

messages resolve last (broken)
  effect widgets=4 surface=false  -> bails, no observer attached
  (no further measurement for the rest of the session)
```

A Playwright probe at 900px reports `{ choiceStage: 0, desktopRails: 2 }` on current `staging`, against `{ choiceStage: 1, desktopRails: 0 }` once the effect re-measures when the surface mounts.

## Environment

- Marinara Engine version: 2.4.4 (`staging`, currently `5c6ac775e`)
- Install type: source
- OS + version: Windows 11
- Browser or app shell: Chromium and WebKit via the Playwright lanes; the width is what matters, not the engine

## Logs, screenshots, or video

No console errors. The surface renders completely and only the layout branch is wrong, which is why it reads as a styling problem rather than a failure.

## Additional context

Found while investigating #5626. The two tests named there are now green on `staging` after #5633 and #5618, but both of those addressed the symptoms; this is the app-side defect underneath, and it survives on any path where the compact decision depends on measurement instead of the `window.innerWidth < 768` shortcut.

Workaround for users: resize the window once after opening the chat.

## Template check

Please **uncheck (untick)** the box below before submitting so we know you read the template. It is intentionally pre-checked:

- [ ] I DID NOT read this template and provide the requested details.

### PR DESCRIPTION
**Pull request by Romuromylus**
## Linked issue

Closes #5654

## Why this change

This started as a fix for #5626's two flaky mobile WebKit tests. While it was in review, #5633 and #5618 landed and stabilised both of those tests on `staging`, so I have dropped that work and re-scoped this PR onto the app-side defect the investigation turned up, which is still live.

`GameSurface` decides its compact HUD layout in an effect keyed on `[combatUiActive, normalizedWidgets.length]`. That effect measures `hudSurfaceRef.current` and attaches a `ResizeObserver` to it. But `GameSurface` also has an early return for its messages-loading branch, which mounts no surface at all, and the widget state hydrates from chat metadata (`useSyncGameState`) independently of the messages query. When the widget count lands while that branch is showing, `updateWidgetLayout` finds no element and returns, and `const surface = hudSurfaceRef.current` is `null` so no observer is attached either. Neither dependency changes when the real surface mounts a moment later, so nothing re-measures for the rest of the session.

#5618 changed that effect's reset from a flat `false` to `window.innerWidth < 768`. That leaves the stranded value correct on a phone, which is why the mobile CYOA test now passes either way, but it does not address the measurement or the missing observer. Every compact decision that depends on the overlap estimate rather than the width shortcut is still wrong.

The visible result, at a tablet or narrow-desktop width, is that the desktop widget rails render on top of the dialogue and the compact choice stage never appears. Only a window resize recovers it, because the `window` resize listener is still registered and `updateWidgetLayout` re-reads the ref when called.

Instrumenting the effect shows both orderings directly:

```
messages resolve first (correct)
  effect widgets=4 surface=true   -> measured w=900 nextCompact=true

messages resolve last (broken)
  effect widgets=4 surface=false  -> bails, no observer attached
  (no further measurement for the rest of the session)
```

And at 900px on current `staging`:

```
without the fix:  { choiceStage: 0, desktopRails: 2 }
with the fix:     { choiceStage: 1, desktopRails: 0 }
```

## What changed

- `packages/client/src/components/game/GameSurface.tsx`: track the HUD surface node in state alongside the existing ref and add it to the layout effect's dependencies, so mounting the surface re-establishes both the measurement and the `ResizeObserver`. The ref is kept because framer-motion consumes it as `constraintsRef` in three places, and it is still assigned in the same commit-time ref phase, so drag-constraint resolution is unchanged. The messages-loading branch gets a `data-component="GameSurface.MessagesLoading"` marker, matching the convention already used across this file; no runtime selector reads it, every existing `data-component` consumer matches an exact unrelated value.
- `e2e/core-flows.e2e.ts`: one new desktop case at 900px. It holds the first page of messages until the loading branch has committed, which pins the ordering rather than leaving it to runner timing, then asserts the branch was actually observed so the hold cannot silently degrade into a no-op, and finally requires the compact stage to be mounted with no desktop rails. 900px is the point of the case: it is past the `width < 768` shortcut, so the compact layout can only be reached by measuring.
- `CHANGELOG.md`: entry for the fix.

Dropped from the earlier scope: the transcript-swipe change, since #5633 fixed that test upstream with a fresh page per fixture and that approach is better than mine (it does not depend on init-script evaluation order, which Playwright documents as undefined); and the guard I had added to the mobile CYOA test, since #5618's reset makes that case pass with or without this fix, so it no longer demonstrates anything.

## Validation

- [x] `pnpm check` passes locally
- [x] Container (Docker / Podman) built and ran without issue
- [x] Ran the app, clicked through the changes manually
- [x] Checked edge cases (light + dark mode, mobile viewport, empty states, error paths)
- [x] Above manual verification completed (describe below)
- [x] Read and followed `CONTRIBUTING.md`

### Manual verification notes

Being precise about what backs the ticks above, since two of them are Playwright-driven rather than hand-driven:

- `pnpm check` passes (exit 0), as does `pnpm version:check`.
- New tablet case, `--project=desktop-chromium --retries=0`: fails 2/2 with the client fix reverted, reporting the rails rendered and no choice stage, and passes 3/3 with it.
- The hold's fail-open path is covered: pointing its wait at a marker that does not exist fails on `expected the Game messages-loading branch to render while the first page of messages was held` rather than passing quietly or timing out somewhere unrelated.
- Full `mobile-webkit` job run locally earlier in this branch's life: 101 passed, 96 skipped, 0 failed.
- CI is green on this head across every lane: all three `desktop-chromium` shards, all four `mobile-webkit` shards, all three `mobile-chromium` shards, `complete-node-regressions`, `container-build-test`, `pnpm-validate` and CodeQL. The container tick rests on that `container-build-test` job; I did not build an image by hand.
- The "ran the app" and "edge cases" ticks rest on driving the real app through Playwright at the affected widths (900x800 for the fix, plus the full lanes at 1440x900, Pixel 7 and iPhone 15 Pro), not on a hand-driven browser session. Light/dark and empty-state paths are untouched by this diff: it changes one layout branch selection and adds one DOM marker.

Things I would appreciate a second pair of eyes on:

- The fix on a real tablet, or a desktop window dragged to roughly 900px: open a Game chat that has HUD widgets on a slow connection and confirm the widgets end up flanking the choice cards rather than sitting over the dialogue.
- Whether the new case belongs in the desktop lane at all, or whether you would rather this ships without a browser test given #5622's concerns about matrix time. It adds about 10 seconds.

## Docs and release impact

- [ ] No docs changes needed
- [x] Updated docs (README / CONTRIBUTING / android/README / CHANGELOG) as needed
- [ ] Translated docs on the `docs-i18n` branch updated to match, or a `[docs-i18n]` follow-up issue opened (see CONTRIBUTING.md § Translated documentation)
- [ ] Version/release files updated (only if this PR includes a version bump)

## UI evidence (if applicable)

The failure is a layout state rather than an error, so the clearest evidence is the element counts above: at 900px the pre-fix build mounts two `[data-game-widget-rail]` elements and no `GameSurface.MobileChoiceStage`, and the post-fix build mounts the stage and no rails.

## Notes for review

`#5626` can be closed as resolved by #5633 and #5618; its two tests are green on `staging` and nothing in this PR touches them any more. I left it open rather than closing someone else's issue on my own read.

The branch carries a merge of `staging` (the `CHANGELOG.md` conflict kept both entries, and the `core-flows.e2e.ts` conflict was resolved in favour of #5633's version wholesale).

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed Game HUD widget rails overlapping dialogue on tablets and narrow desktop windows.
  * Improved compact HUD layout detection when widgets load asynchronously or while messages are loading.
  * Ensured the HUD updates correctly when its display surface appears after loading.

* **Tests**
  * Added end-to-end coverage for HUD compaction at a 900px viewport.

* **Documentation**
  * Documented the HUD layout fix in the changelog.

### CODE REVIEW

File: e2e/core-flows.e2e.ts [hunk 1]
```diff
@@ -19568,6 +19568,187 @@ test("mobile Game keeps CYOA usable above four HUD widgets", async ({ page, requ
...
+        // is where the widget state hydrates. Frames rather than a fixed delay,
+        // so the wait stretches on a loaded runner instead of expiring early.
+        await page
+          .evaluate(
+            () =>
+              new Promise<void>((resolve) => {
+                requestAnimationFrame(() => requestAnimationFrame(() => resolve()));
+              }),
+          )
+          .catch(() => undefined);
```
[the remark this answers is not in the document]
Comment (Romuromylus) [reply e2e/core-flows.e2e.ts:19717]: Good catch, and it holds up against the code: `page.evaluate` carries no timeout of its own, so a page whose frames are throttled leaves that promise pending, and the `.catch()` cannot recover a promise that never settles. The held request would then sit there until the test-level timeout, which is exactly the opaque failure the named assertion below was added to avoid.

Fixed in 159104144 by racing the frames against a one second ceiling, so frames still win in the normal case and the timer only decides the case where they never arrive.
